csrc/scheduler/pointwise_utils.cpp

```diff
@@ -5,247 +5,34 @@
-getIndexedConsumerToProducerMap(Fusion* fusion, const ComputeAtMap& ca_map) {
-  std::unordered_multimap<
-      std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>,
-      std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
-      indexed_id_map;
-
-  for (auto expr : fusion->exprs()) {
-    if (auto gather = dynamic_cast<TorchGatherOp*>(expr)) {
-      auto p_id = gather->getIndexedID();
-      auto c_id = gather->getConsumerOfIndexedID();
-      indexed_id_map.emplace(
-          ca_map.disjointSetOf(c_id, IdMappingMode::EXACT),
-          ca_map.disjointSetOf(p_id, IdMappingMode::EXACT));
-    } else if (auto index_select = dynamic_cast<IndexSelectOp*>(expr)) {
-      auto p_id = index_select->getIndexedID();
-      auto c_id = index_select->getConsumerOfIndexedID();
-      indexed_id_map.emplace(
-          ca_map.disjointSetOf(c_id, IdMappingMode::EXACT),
-          ca_map.disjointSetOf(p_id, IdMappingMode::EXACT));
-    } else {
-      // Note there's no consumer ID for select. This means we can't
-      // just propagate from consumers to indexed producers. It seems
-      // it's necessary to schedule producers and consumers separately
-      // in those cases.
-      continue;
-    }
-  }
-
-  return indexed_id_map;
-}
-
-// Check if a root ID of a fusion input tensor that is indirectly
-// accessed by ops such as torchGather needs to be mapped with
-// a reference tensor. Select has a similar effect as squeeze as the
-// indexed domain is removed, so the domain does not need to be mapped
-// as long as the tensor is a fusion input. Similarly, in index_select
-// and torchGather, if the output domain is a broadcast, it does not
-// need to be mapped if not resolved.
-bool canIgnoreIndexedInputDomainID(
-    TensorView* input_tv,
-    IterDomain* root_id,
-    const ComputeAtMap& ca_map) {
-  NVF_ERROR(input_tv->isFusionInput());
-  for (auto use : input_tv->uses()) {
-    if (auto select = dynamic_cast<SelectOp*>(use)) {
-      if (root_id != select->getIndexedID()) {
-        return false;
-      }
-    } else if (auto index_select = dynamic_cast<IndexSelectOp*>(use)) {
-      // If the root_id is an indexed ID, and the consumer ID may be a
-      // broadcast. In that case, nothing needs to be mapped if the
-      // consumer broadcast is not resolved
-      if (root_id != index_select->getIndexedID() ||
-          !ca_map
-               .getConcreteMappedID(
-                   index_select->getConsumerOfIndexedID(),
-                   IdMappingMode::PERMISSIVE)
-               ->isBroadcast()) {
-        return false;
-      }
-    } else if (auto gather = dynamic_cast<TorchGatherOp*>(use)) {
-      // TODO: Remove this. Once slice is used for torchGather, this
-      // should not be necessary. For now, it is necessary to not
-      // break the existing torchGather tests
-      if (!gather->exactSizes()) {
-        continue;
+TensorView* DomainMap::findReferenceTensor(int64_t minimum_num_axes) const {
```

Comment:
nitpick: do we still want to keep pointwise_utils.cpp at this point. the majority of stuff is moved domain_map.cpp.

We can move these inside pointiwse.cpp directly like how transpose scheduler is handling this..

Reply:
I don't have any particular preference